cis194/week1/tien/homework1/src/Lib.hs

```diff
@@ -0,0 +1,30 @@
+module Lib
+  ( toDigits
+  , toDigitsRev
+  , buildWeights
+  , doubleEveryOther
+  , sumDigits
+  , validate
+  ) where
+
+-- Exercise One: --
+toDigits :: Integer -> [Integer]
+toDigits num
+  | num > 0 = map (read . (: [])) $ show num :: [Integer]
+  | otherwise = []
+
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev num
+  | num > 0 = reverse $ toDigits num
+  | otherwise = []
+
+-- Exercise Two: --
+buildWeights :: Integer -> [Integer]
+buildWeights num = reverse $ map ((+ 1) . (`mod` 2)) [0 .. num - 1]
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther arr = zipWith (*) arr $(buildWeights . toInteger . length) arr
+
+-- Exercise Three: --
+sumDigits :: [Integer] -> Integer
+sumDigits arr = sum $ map sum $ map (toDigits) arr
```

Comment:
```suggestion
sumDigits arr = sum . map sum . map toDigits $ arr
```

Also, have a look at `concatMap` see if you can improve this line of code using it.